extension/src/state_aggregate.rs

```diff
@@ -738,19 +738,28 @@ impl CompressedStateAggTransState {
 fn duration_in_inner<'a>(
     state: Option<StateEntry>,
     aggregate: Option<CompressedStateAgg<'a>>,
-    range: Option<(TimestampTz, Option<crate::raw::Interval>)>,
+    range: Option<(i64, Option<i64>)>, // start and interval
 ) -> crate::raw::Interval {
     let time: i64 = if let Some((start, interval)) = range {
-        let (start, end) = if let Some(interval) = interval {
-            let interval = crate::datum_utils::interval_to_ms(&start, &interval);
+        let end = if let Some(interval) = interval {
             assert!(interval >= 0, "Interval must not be negative");
-            let start = start.into();
-            (start, start + interval)
+            start + interval
         } else {
-            (start.into(), i64::MAX)
+            i64::MAX
         };
         assert!(end >= start, "End time must be after start time");
         if let (Some(state), Some(agg)) = (state, aggregate) {
+            assert!(
+                !agg.0.compressed,
+                "unreachable: interval specified for compressed aggregate"
+            );
+            assert!(
+                start >= agg.0.first_time,
+                "Start time ({}) cannot be before first state ({})",
+                start,
+                agg.0.first_time,
+            );
```

Comment:
I'm not sure if it's more reasonable to assert here, or to simply exclude the interval before `first_time`.

Reply:
I removed this assert; I think excluding the prior interval is a better choice here.